### DISCUSSION TITLE
NEXTJS 13: self.__next_f.push append to <script> tag in source code

### DISCUSSION THREAD

**Discussion by mmsafari** [vercel/next.js#58487] [Help] [answered]
Hello, somebody knows what it is.__next_f.push in page source code??

i use nextjs **app-router** (v13.5.5)

The data of the previous pages is appended to the source code as **self.__next_f.push** in **<script>** tags. self.__next_f.push double all content on the webpage. Thus, the page becomes almost twice as large.
I have not seen this in version 12 or 13 with next.js **pages router**.

notice: i update our nextjs version to v14.0.2 and the issues still exists.

![Capture](https://github.com/vercel/next.js/assets/48034022/d6545baf-02aa-4e47-ac27-3ec49ff0ab99)

**Comment by amirjalalli** [comment]
I have same problem

**Comment by thaophamvan** [comment]
I have the same issue. It should not in source code because html already pre-render. We need an answer from Vercel team
![image](https://github.com/vercel/next.js/assets/11343747/70d9c909-40be-4c06-a8ca-672d0025b766)

**Comment by leo-cheron** [comment]
Duplicate of https://github.com/vercel/next.js/discussions/42170

**Comment by leerob** [comment, marked as answer]
https://github.com/vercel/next.js/discussions/42170#discussioncomment-8137079
